File: bench/dispatch.py
# The Python twin of dispatch.sl, written as `match` -- which Python 3.10 grew and which is the same
# construct slate has, a subject compared against literal patterns in order.


def kind(w):
    match w:
        case "add":
            return 1
        case "sub":
            return 2
        case "mul":
            return 3
        case "div":
            return 4
        case "mod":
            return 5
        case _:
            return 0


def run(words):
    total, i = 0, 0

    while i < 5000000:
        total = total + kind(words[i % 6])
        i = i + 1

    return total


print(run(["add", "sub", "mul", "div", "mod", "nope"]))

File: bench/dispatch.php
<?php
// The PHP twin of dispatch.sl, written as `match` -- which PHP 8 grew, and which is the same
// construct slate has: a subject compared against literal patterns, answering a value.
//
// **PHP compiles a `match` whose arms are all literal strings to a HASH lookup into a jump table**,
// so this is the "one that could hash the subject once" dispatch.sl's header anticipates.

function kind($w)
{
    return match ($w) {
        "add" => 1,
        "sub" => 2,
        "mul" => 3,
        "div" => 4,
        "mod" => 5,
        default => 0,
    };
}

function run($words)
{
    $total = 0;
    $i = 0;

    while ($i < 5000000) {
        $total = $total + kind($words[$i % 6]);
        $i = $i + 1;
    }

    return $total;
}

echo run(["add", "sub", "mul", "div", "mod", "nope"]), "\n";
